### ISSUE TITLE
cluster_label_prop() with directed graph

### ISSUE THREAD

**Issue by alexandergerber** [igraph/rigraph#214]
If a directed graph is passed to `cluster_label_prop()` together with a vector of initial states containing negative values,  R crashes. I guess it would be better if the  function whould throw an error if a directed graph is passed. 

Here is a reproducible example:

```
library(igraph)

adj_matrix <- structure(c(0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 
                          0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 
                          0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 
                          1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 
                          0), .Dim = c(9L, 9L), 
                        .Dimnames = list(c("a", "b", "c", "d", "e", "i", "g", "f", "h"), 
                                         c("a", "b", "c", "d", "e", "i", "g", "f", "h")))
graph <- graph_from_adjacency_matrix(adj_matrix)

init <- c(1,-1,-1,-1,-1,-1,-1,-1,2)

#R crashes
com1 <- cluster_label_prop(graph, initial = init, fixed = init %in% c(1,2) )

#Works fine
com2 <- cluster_label_prop(as.undirected(graph), initial = init, fixed = init %in% c(1,2) )
 
```

**Comment by szhorvat** [comment]
This is fixed in the current release, but there's good reason to verify it in the dev version. Could you do that please @vtraag ?

In the current release:

```
> com1 <- cluster_label_prop(graph, initial = init, fixed = init %in% c(1,2) )
Modularity is implemented for undirected graphs only.
```

**Comment by szhorvat** [comment]
Actually this is likely the same as https://github.com/igraph/igraph/issues/1852 for which I submitted a PR today.

**Comment by ntamas** [comment]
Issue fixed in the C core; it will be fixed in the R interface in version 1.3 when we update the C core to 0.9.5

**Comment by szhorvat** [comment]
The dev version of 1.3 can be installed using `remotes::install_github("igraph/rigraph@master")` (if you have the R development tools installed). I verified that this no longer crashes.
